src/1-raw-data/mailinglists/main.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This package is for loading different mailing list data types into Cloud Storage.
+*/
+
+package main
+
+//TODO
+// Check the most recent file stored and pull only what isn't there
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"flag"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"io"
+	"log"
+	"net/http"
+)
+
+type storageConnection struct {
+	ctx    context.Context
+	client *storage.Client
+	bucket *storage.BucketHandle
+}
+
+var (
+	projectID      = flag.String("project-id", "", "project id")
+	bucketName     = flag.String("bucket-name", "", "bucket name to store files")
+	mailingList	   = flag.String("mailinglist", "piper", "Choose which mailing list to process either piper (default), mailman")
+	mailingListURL = flag.String( "mailinglist-url", "", "mailing list url to pull files from")
+	startDate 	   = flag.String("start-date", "", "Start date in format of year-month-date and 4dig-2dig-2dig")
+	endDate	       = flag.String( "end-date", "", "End date in format of year-month-date and 4dig-2dig-2dig")
+	gcs            = storageConnection{}
```

Comment:
It doesn't seem like you have a real need for this to be a global variable so consider moving this into `main`

Reply:
Is the recommendation to move all variables into main and pass them to the functions vs making them accessible globably?

Reply:
Moved gcs but left rest as global. Will discuss offline.